\documentclass[12pt]{article}

\usepackage{fixltx2e}
\usepackage{textcomp}
\usepackage{fullpage}
\usepackage{amsfonts}
\usepackage{verbatim}
\usepackage[english]{babel}
\usepackage{pifont}
\usepackage{color}
\usepackage{setspace}
\usepackage{lscape}
\usepackage{indentfirst}
\usepackage[normalem]{ulem}
\usepackage{booktabs}
% \usepackage{nag}
\usepackage{natbib}
% \usepackage{bibtex}
\usepackage{float}
\usepackage{latexsym}
\usepackage{hyperref}
\usepackage{url}
% \usepackage{html}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{array}
%\usepackage{mhchem}
\usepackage{ifthen}
\usepackage{caption}
\usepackage{xcolor}
\usepackage{amsthm}
\usepackage{amstext}
\usepackage{nicefrac}
\usepackage{algorithm}
\usepackage{algorithmic}
\usepackage[scientific-notation=true]{siunitx}
\usepackage{subfigure}
\usepackage[flushleft]{threeparttable}
\usepackage{lineno}
\usepackage{adjustbox}
\usepackage{ragged2e}
\usepackage{authblk}
\usepackage{multirow}

\setlength{\parskip}{1em}
\renewcommand{\baselinestretch}{2.0}
\renewcommand\Affilfont{\small}

\begin{document}

\title{StarBEAST2 brings faster species tree inference and accurate estimates of substitution rates}
\author[1,2]{Huw A. Ogilvie\thanks{huw.ogilvie@anu.edu.au}}
\author[2,3]{Alexei J. Drummond}
\affil[1]{Division of Evolution, Ecology and Genetics, Research School of Biology, Australian National University, Canberra, Australia}
\affil[2]{Centre for Computational Evolution, University of Auckland, Auckland, New Zealand}
\affil[3]{Department of Computer Science, University of Auckland, Auckland, New Zealand}

\maketitle

\clearpage

\justifying

\section*{Abstract}

StarBEAST2 is available through the BEAUTi package manager in BEAST 2.4 and above.

Keywords: Phylogenetic methods, substitution rate, relaxed clock, multispecies coalescent, concatenation.

\section*{Introduction}

Talk about Ogilvie 2016 a lot...

\section*{New Approaches}

\subsection*{Analytical integration of population sizes}

MCMC numerically integrates over many parameters to produce explicit estimates
of those parameter values. From a researcher's perspective, some of these may be
``nuisance'' parameters not of scientific interest. For example species tree
topology and divergence times may be of interest, but not population sizes. For
analytically tractical parameters, an analytical solution will integrate over
the entire range of values at each MCMC step, and may be more performant than
numerical integration. However explicit estimates will not be produced and this
integration is only suitable for nuisance parameters. Among-site rate variation
is already analytically integrated; the likelihood of each site is calculated
for all possible discrete gamma rates at each step, so the individual site rates
are not estimated \citep{Yang1994}.

Analytical integration of constant per-branch population sizes was first
implemented as part of BEST \citep{EVO:EVO414}. The analytical solution, which
we have added to StarBEAST2, uses an inverse gamma conjugate prior for
population sizes. By default StarBEAST2 fixes the shape of the distribution
$\alpha = 2$ and only estimates the scale $\beta$. Because the mean of an
inverse gamma distribution equals $\nicefrac{\beta}{\alpha - 1}$, under default
settings the hyperprior parameter $\beta$ is an estimate of mean (effective)
population size.

\subsection*{Coordinated tree topology changing operators}

One approach to improving the performance of fully Bayesian multispecies
coalescent analyses which simultaneously estimate gene and species trees is to
develop MCMC operators which propose coordinated changes to both the species
tree and the gene trees in the same step. \cite{Yang01122014} and
\cite{2015arXiv151203843R} introduced Metropolis-Hastings (MH) operators which make
NNI and SPR changes to the species tree topology, and simultaneously make
changes to the gene tree topologies that ensure compatibility of the gene trees
within the proposed species tree. We have reimplemented those operators as a new
operater in StarBEAST2 called ``CoordinatedExchange''.
\cite{2015arXiv151203843R} also describe a proposal distribution which favours
topological changes on shorter branches, and also less radical changes in
topology. StarBEAST2 implements a simpler proposal distribution but still
favours less radical changes by applying adjustable proposal probability
weights to (less radical) NNI moves and (more radical) SPR moves.

\subsection*{Coordinated node height changing operators}

A novel class of coordinated Metropolis operators was introduced by
\cite{Jones010199}. These operators change the height of a non-root non-leaf
species tree node, and the heights of ``connected components'' of gene tree
nodes, by an amount $\epsilon$ chosen from a uniform distribution.
The lower bound of the uniform distribution is the negative length of
the shortest child branch of a connected component or of the species tree node,
and the upper bound is the positive length of the shortest parent branch. As
long as the connected components are chosen with reference only to the topology
of the species tree, the topology of the gene trees, and the mapping of
sampled individuals to species, operators of this class are symmetric
\citep{Jones010199}.

We have developed a new operator called ``CoordinatedUniform'' that belongs to
this class. Individuals from extant species which descend from a species tree
node, or are directly descended from a gene tree node, are referred to as
descendant individuals. The gene tree nodes selected by this operator are those
for which (1) at least one descendant individual is also a descendant individual
of each direct child of the selected species tree node and (2) all descendent
individuals are also descendent individuals of the selected species tree node.

We have also developed a new adaptive MH \citep{Andrieu2008} operator called
``CoordinatedExponential'' which changes the height of the species tree root and
the height of connected components of gene tree nodes by an amount $\epsilon$.
The changed gene tree nodes are those for which at least one descendant individual
is also a descendant individual of each direct child of the species tree root.
Because the length of parent branches of the species tree root or connected
components will be undefined, a different method must be used to choose
$\epsilon$ compared to CoordinatedUniform.

First the lower bound of the species tree root height is defined as the current
height minus the length of the shortest child branch of a connected component or
of the species tree root. The difference between the lower bound and the current
root height is referred to as $x$, and a new random value $x'$ is chosen from an
exponential distribution. The value of $x' - x$ is then used for $\epsilon$. The
median of the exponential distribution is adaptively modified over the course of
an MCMC chain to equal the posterior expectation of $x$.

Because the proposal distribution for the new species tree root height $x'$ is
independent of the current height $x$, the Hastings ratio which is usually
$\nicefrac{q(x', dx)}{q(x, dx')}$ can be simplified to $\nicefrac{q(x')}{q(x)}$.
The (log) Hastings ratio may then be derived from the respective
exponential probability densities of $x$ and $x'$:

\begin{align}
\frac{q(x')}{q(x)} &= \frac{\lambda e^{-\lambda x'}}{\lambda e^{-\lambda x}} = \frac{e^{-\lambda x'}}{e^{-\lambda x}}\\
\therefore \ln\left(\frac{q(x')}{q(x)}\right) &= \ln\left(e^{-\lambda x'}\right) - \ln\left(e^{-\lambda x}\right)\\
& = \lambda x \cdot \ln\left(e\right) - \lambda x' \cdot \ln\left(e\right)\\
& = \lambda \left(x - x'\right) = \lambda \epsilon
\end{align}

\subsection*{Species tree relaxed clocks}

While *BEAST has been sucessfully used with relaxed clocks
\citep{Berv2014120,Lambert2015146}, this involves estimating per-branch
substitution rates (branch rates) separately for each branch of each gene tree.
Gene tree relaxed clock models may accomodate variation in substitution rates
between species branches, but do not produce estimates of species branch rates. To enable
inference of species branch rates, we have developed a new species tree relaxed
clock model.

The challenge of applying a relaxed clock to the species tree is that
phylogenetic likelihood calculations require branch rates for each branch of
each gene tree. Our clock model uses the total expected number of substitutions
$\Sigma \mathbb{E}(S)$ accumulated by a gene branch through all containing
species branches. Substitutions are expected to be accumulated at the mean
clock rate of the gene tree $c$, for example 0.01 for hominoid primate
mitochondrial DNA \citep{doi:10.1146/annurev.es.18.110187.001413}, multiplied
by the lengths of time $L$ spent traversing each species tree branch, multiplied
by the rates $R$ of the corresponding species tree branches
(Table~\ref{tab:branchRateModel}).

\begin{table}[htb!]
\centering
\caption{Expected numbers of substitutions under a species tree relaxed clock.}
\label{tab:branchRateModel}
\begin{threeparttable}
\begin{adjustbox}{center}
\begin{tabular}{|c|c|c|c|c|c|c|c|c|c|c|c|c|}
\hline
\multirow{2}{*}{Branch} & \multirow{2}{*}{$c$} & \multicolumn{3}{c|}{$L$} & \multicolumn{3}{c|}{$R$} & \multicolumn{3}{c|}{$\mathbb{E}(S) = c\cdot L\cdot R$} & \multirow{2}{*}{$\Sigma \mathbb{E}(S)$}\tabularnewline
\cline{3-11}
 & & A & B & AB & A & B & AB & A & B & AB & \tabularnewline
\hline
a & \multirow{2}{*}{0.01} & 1.0 & 0.0 & 0.5 & \multirow{2}{*}{0.7} & \multirow{2}{*}{1.0} & \multirow{2}{*}{1.3} & 0.0070 & 0.0000 & 0.0065 & 0.0135\tabularnewline
\cline{1-1} \cline{3-5} \cline{9-12}
b & & 0.0 & 1.0 & 0.5 & & & & 0.0000 & 0.0100 & 0.0065 & 0.0165\tabularnewline
\hline
\end{tabular}
\end{adjustbox}
\end{threeparttable}
\end{table}

The gene tree branch rates $r$ can then be derived by dividing the total
expected number of substitutions by the total length of that branch $l$. The
gene tree branch rates for the illustrated example
(Figure~\ref{fig:branchRateModel}; Table~\ref{tab:branchRateModel}) are
therefore:

\begin{align}
r_a &= \frac{\Sigma \mathbb{E}(S_a)}{l_a} = \frac{0.0135}{1.5} = 0.009\\
r_b &= \frac{\Sigma \mathbb{E}(S_b)}{l_b} = \frac{0.0165}{1.5} = 0.011
\end{align}

\begin{figure}[htb!]
\centering
\includegraphics[width=9cm]{relaxed_clock.pdf}
\caption
{Example multispecies coalescent phylogeny to illustrate species tree relaxed
clocks. There are two extant species ``A'' and ``B'', and one ancestral species ``AB''.
Within the species tree there is a single gene tree with extant individuals ``a''
and ``b''. The single speciation event occurs at time T1, and the single coalescent
event occurs at time t1.}
\label{fig:branchRateModel}
\end{figure}

The new species tree relaxed clock model is available in StarBEAST2. At present
only the well-established uncorrelated log-normal (UCLN) branch rate model
\citep{10.1371/journal.pbio.0040088} is supported, but there is no theoretical
or practical barrier to implementing support for other types of branch rate
models in the future.

\section*{Results}

\subsection*{StarBEAST2 correctly implements the multispecies coalescent}

New methods must be shown to be mathematically correct implementations of a
given model. One way to accomplish this for MCMC methods is to estimate
parameters from a prior distribution using the MCMC kernel, and to also draw
independent samples from an identical distribution by simulation. The
resulting parameter distributions should also be identical if the implementation is
correct. We used this method to test the correctness of the novel features in
StarBEAST2; analytical population size integration, coordianted operators, and
species tree relaxed clocks. Simulated and StarBEAST2 estimated distributions
were identical for species and gene tree topologies (Figure~S1,S2), species and
gene tree node heights (Figure~S3,S4), and for gene tree branch rates (Figure~S5).
This combination of results supports the mathematical correctness of StarBEAST2.

\subsection*{\textit{Pseudacris} chorus frogs have intermediate coalescent branch lengths}

To characterize the performance of coordinated operators, methods of population size
integration and relaxed cocks, we tested StarBEAST2 using real sequence data.
The data set used for this analysis is from the North American chorus frog genus
\textit{Pseudacris}, and was originally collected and analyzed by
\cite{Barrow201478}. A key metric of phylogenies that can be used to judge
whether it is necessary to employ multispecies coalescent models is the average
branch length in coalescent units $\nicefrac{\tau}{2Ne}$. Given short branch
lengths, concatenation is unable to infer accurate species trees regardless of
the number of loci used, but for long branch lengths, concatenation is
approximately as accurate as *BEAST \citep{Ogilvie01052016}. Using StarBEAST2,
the average branch length within this genus was determined to be
$\nicefrac{1.69\tau}{2Ne}$. This is an intermediate average length, longer than
the shallow simulations analyzed by \cite{Ogilvie01052016} which had short
branch lengths of $\nicefrac{0.54\tau}{2Ne}$.

\subsection*{Coordinated operators and analytical integration improve computational performance}

To determine which configuration of new features would achieve the fastest
performance, we ran StarBEAST2 using different combinations of operators,
methods of population size integration and different relaxed clocks. To measure
convergence both effective sample size (ESS) per hour and ESS per million states
were computed for each independent chain. ESS per hour can be used calculate the
total time required for a converged chain (where ESS equals or exceeds 200), and
reflects how effectively operators explore the space of trees and parameters,
and the computational time required by each operator proposal and likelihood
calculation. In contrast, ESS per million states reflects only the exploration
of tree and parameter space independently of calculation times. A variety of
statistics were logged for each analysis, but the height of the species tree had
the slowest average convergence rates so we used this statistic to judge
computational performance (Figure~S7-Sx).

Coordinated topology operators consistently reduced ESS per hour,
regardless of the addition of height changing operators or the method of
population size integration (Figure~\ref{fig:realEssPerHour}). However these
operators had little effect on ESS per million states (Figure~\ref{fig:realEssPerMstates}).
The latter result suggests that coordinated topology operators are no more
effective than na\"ive operators at proposing new states, and the former result
shows how the increased complexity of these operators slows convergence because
of their longer calculation times.

\begin{figure}[htb!]
\centering
\includegraphics[height=14cm]{speciesTreeHeight_ess_per_hour.pdf}
\caption
{Effect of operators, population size integration and clock models on
convergence. The performance of every combination of settings was measured when
applied to 22 \textit{Pseudacris} loci. The bar marked with $\ast$ represents *BEAST
settings. Topology operators refers to the
replacement of na\"ive NNI and SPR operators with coordinated operators. Heights
operators refers to the addition of operators which make coordinated changes to
internal node heights. Uncorrelated log-normal (UCLN) relaxed clocks were applied
to either each gene tree or to the species tree. Trimmed means of species tree
height effective sample size (ESS) per hour were calculated from 32 independent
chains. 25\% trim was used to reduce the influence of outliers. All error bars
are 95\% confidence intervals calculated by bootstrapping.}
\label{fig:realEssPerHour}
\end{figure}

\begin{figure}[htb!]
\centering
\includegraphics[height=14cm]{speciesTreeHeight_ess_per_mstates.pdf}
\caption
{Effect of operators, population size integration and clock models on effective
sample size (ESS) per million states. The performance of every combination of
settings was measured when applied to 22 \textit{Pseudacris} loci. The bar marked with $\ast$ represents *BEAST
settings. Topology operators refers to the
replacement of na\"ive NNI and SPR operators with coordinated operators. Heights
operators refers to the addition of operators which make coordinated changes to
internal node heights. Uncorrelated log-normal (UCLN) relaxed clocks were applied
to either each gene tree or to the species tree. Trimmed means
of species tree height ESS per million states were calculated from 32
independent chains. 25\% trim was used to reduce the influence
of outliers. All error bars are 95\% confidence intervals calculated by
bootstrapping.}
\label{fig:realEssPerMstates}
\end{figure}

Coordinated height changing operators consistently improved convergence both in
terms of ESS per hour and ESS per million states
(Figure~\ref{fig:realEssPerHour},\ref{fig:realEssPerMstates}). Both species tree
and gene tree relaxed clock analyses benefitted from enabling those operators.
Analytical integration of population sizes only improved the computational
performance of gene tree relaxed clock analyses; unlike height changing
operators it did not change the performance of species tree relaxed clock analyses.

Convergence of species tree relaxed clock analyses was largely slower than
convergence of gene tree relaxed clock analyses. The relative performance in ESS
per hour was worse than the relative performance in ESS per million states
(Figure~\ref{fig:realEssPerHour},\ref{fig:realEssPerMstates}), which shows that
compared to gene tree relaxed clock analyses the likelihood calculations have a
higher time cost, but also that operators become less effective at
exploring tree and parameter space.

\subsection*{StarBEAST2 is approximately three times faster than *BEAST}

Our reanalysis of \textit{Pseudacris} sequence data shows that analytical
integration of population sizes and coordinated height changing operators both
improve convergence. To show that this increased performance is generally
applicable, and to demonstrate that StarBEAST2 can accurately reconstruct branch
rates under a species tree relaxed clock model, we needed to apply StarBEAST2 to
many different data sets for which the true species tree is known. To accomplish
this we simulated 64 species trees of 19 taxa each under a birth-death model
with log-normally distributed branch rates. Gene trees evolving within each
species tree were simulated according to the multispecies coalescent model with
log-normally distributed mean clock rates. Finally, sequences were simulated
according to an HKY process for each gene \citep{Hasegawa1985, Goldman1993}. The
parameters used for these simulations were chosen to also produce intermediate
branch lengths in coalescent units; the average simulated branch length was
$\nicefrac{1.44\tau}{2Ne}$.

Simulated data was analysed using BEAST with concatenation and with StarBEAST2.
The different settings used for StarBEAST2 were *BEAST settings, performant
settings with gene tree relaxed clocks, and performant settings with species
tree relaxed clocks. *BEAST settings matched the configuration of *BEAST prior
StarBEAST2; explicit MCMC integration of population sizes, no coordinated
operators, na\"ive NNI and SPR topology operators, and a UCLN relaxed clock
applied to each gene tree. Performant settings included analytical integration of
population sizes, coordinated height changing operators, and na\"ive NNI and SPR
topology operators.

Again the parameter with the slowest convergence given any multispecies coalescent
settings was the height of the species tree. Based on that parameter, the
average performance increase using StarBEAST2 with performant settings over
*BEAST settings was 3.2 times, with a 95\% confidence interval between 2.6 and
3.8 (Figure~\ref{fig:simulatedEssPerHour}). The slowest converging parameter for
concatenated analyses was the posterior probability, and when using the same number
of loci convergence of the posterior statistic using concatenation was
114 times faster than the convergence of species tree height
using StarBEAST2 with *BEAST settings. When concatenation was used with ten
times the number of loci, the performance difference was minimal --- 1.2 times,
with a confidence interval between 1.0 and 1.5 (Figure~\ref{fig:simulatedEssPerHour}).

Unlike \textit{Pseudacris}, using a species tree relaxed clock with
performant settings was faster than *BEAST settings. However, this was still
significantly slower than using gene tree relaxed clocks, as the average
performance increase was smaller at 1.5 times with a confidence interval between
1.2 and 1.9 (Figure~\ref{fig:simulatedEssPerHour}).

\begin{figure}[htb!]
\centering
\includegraphics[width=16cm]{multiple_ess_per_hour.pdf}
\caption
{Convergence of different methods applied to simulated data. Methods are
concatenation with 22 loci, concatenation with 220 loci (10X), StarBEAST2 with
*BEAST settings and 22 loci, and StarBEAST2 with performant settings, 22 loci,
and uncorrelated log-normal relaxed clocks applied to the gene trees (GT-UCLN) or
to the species tree (ST-UCLN). Relative ESS per hour is the trimmed mean of
effective sample size per hour for each replicate divided by the species tree height estimated using *BEAST
settings. 25\% trim was used to reduce the influence of
outliers. All error bars are 95\% confidence intervals calculated by
bootstrapping. N = 64.}
\label{fig:simulatedEssPerHour}
\end{figure}

\subsection*{Relative accuracy varies depending on metric}

The accuracy of StarBEAST2 relative to concatenation varied depending on the
metric and whether equal numbers of loci were used. Relative species tree error
measures the accuracy of estimated branch lengths; for that metric StarBEAST2
using 22 loci outperformed concatenation using 22 loci, and matched the
performance of concatenation using 220 loci
(Figure~\ref{fig:speciesTreeError}A). Pendant edge bias measures systematic bias
in the estimated ages of extant species; for that metric StarBEAST2 was much
less biased than concatenation regardless of the number of loci
(Figure~\ref{fig:speciesTreeError}B). Rooted Robinson-Foulds distances measure
the topological accuracy of estimated trees; for that metric concatenation using
22 loci was almost as accurate as StarBEAST2, and much more accurate when using
220 loci (Figure~\ref{fig:speciesTreeError}C). For no metric did the choice of
gene tree or species tree relaxed clocks affect the accuracy of StarBEAST2
(Figure~\ref{fig:speciesTreeError}A,B,C).

\begin{figure}[htb!]
\centering
\includegraphics[width=6.5cm]{tree_error.pdf}
\caption
{Accuracy of different methods applied to simulated data. Methods are concatenation with 22 loci, concatenation with 220 loci
(10X), StarBEAST2 with *BEAST settings and 22 loci, and StarBEAST2 with
performant settings, 22 loci, and uncorrelated log-normal relaxed clocks applied
to the gene tree (GT-UCLN) or to the species tree (ST-UCLN). (A) Trimmed mean of
relative species tree error, a measure of branch length error. (B) Trimmed
mean of mean pendant edge bias, which measures biased estimates of the ages of
extant species. (C) Trimmed mean of mean rooted Robinson-Foulds distances, a
measure of topological error. 25\% trim was used to reduce the
influence of outliers. All error bars are 95\% confidence intervals calculated
by bootstrapping. N = 64.}
\label{fig:speciesTreeError}
\end{figure}

\subsection*{Concatenation cannot accurately infer branch rates given intermediate branch lengths}

While the convergence of species tree relaxed clocks analyses was slower than
gene tree relaxed clock analyses using StarBEAST2, species tree relaxed clocks
enable inference of per-branch substitution rates (branch rates) under the multispecies
coalescent model. To gauge the accuracy of estimated branch rates, we used
a simple linear regression (SLR) with the true rate of each simulated branch as
the response variable, and the posterior expectation of the rate of that branch
(conditional on the corresponding clade being monophyletic in the posterior
samples) as the explanatory variable (Figure~\ref{fig:branchRatesLM}). If all
estimates equal the truth, then the $R^2$ correlation coefficient of the SLR
will equal 1, the slope of the SLR will also equal 1, and the intercept will
equal zero.

\begin{figure}[htb!]
\centering
\includegraphics[width=16cm]{branch_rates.pdf}
\caption
{Recovery of species tree branch rates using concatenation and StarBEAST2.
Methods are concatenation with 22 loci, concatenation with 220 loci (10X), and
StarBEAST2 with performant settings, 22 loci, and an uncorrelated log-normal
relaxed clocks applied to the species tree (ST-UCLN). Estimated rates are the
posterior expectation of the branch rate. $R^2$ values and lines of best fit were calculating using a separate
simple linear regression for each method. Root branch rates, which were fixed at
1.0, were excluded. N = 64.}
\label{fig:branchRatesLM}
\end{figure}

Using either 22 or 220 loci, concatenation was unable to correctly estimate
branch lengths --- not only did $R^2 = 0.03$ and $0.04$ respectively, but the
slope and intercept were $0.32x + 0.68$ and $0.79x + 0.21$ respectively. By
applying a UCLN relaxed clock to the species tree, StarBEAST2 can be used to
infer many branch rates --- the $R^2$ correlation was much stronger at $0.15$,
and the slope and intercept were $0.93$ and $0.07$ respectively, close to
perfect values.

\clearpage

\section*{Discussion and Conclusions}

More extreme values expected because of SPILS?

StarBEAST2 is open source and free software, its source code and development
history is available through GitHub
(\url{https://github.com/genomescale/starbeast2}).

\section*{Materials and Methods}

For all StarBEAST2 and concatenation analyses, the version of BEAST used was v2.4.0.
For all simulations, the version of biopy used was 0.1.9.

\subsection*{Mathematical correctness of StarBEAST2}

Simulated trees were generated using biopy \citep{biopy}, and trees sampled from
a prior distribution were generated using StarBEAST2 with all new features
enabled. This included analytical integration of population sizes, coordinated
tree topology and node height changing operators, and a species tree relaxed
clock. 100,000 species trees were simulated, and 100,000 trees were sampled from
the prior at a rate of one every 1000 after a 10\% burnin period. Identical
parameters were used for the simulation and for the StarBEAST2 run including the
prior distributions (Table~\ref{tab:correctParameters}).

\begin{table}[htb!]
\centering
\caption{Parameters used for verifying mathematical correctness.}
\label{tab:correctParameters}
\begin{threeparttable}
\begin{adjustbox}{center}
\begin{tabular}{|r|l|}
\hline
Number of species & 5\tabularnewline
\hline
Birth rate & Fixed at 200.0\tabularnewline
\hline
Death rate & Fixed at 100.0\tabularnewline
\hline
Species tree branch rates & Log-normal ($\mu* = 1.0, \sigma = 0.16$)\tabularnewline
\hline
Haploid population sizes & Inverse gamma ($\alpha = 2.0, \beta = 0.002$)\tabularnewline
\hline
Haploid individuals per species & 1\tabularnewline
\hline
Number of gene trees & 2\tabularnewline
\hline
Per-locus mean clock rates & 0.5, 2.0\tabularnewline
\hline
\end{tabular}
\end{adjustbox}
\begin{tablenotes}
\small
\item {*}: Mean in real space.
\end{tablenotes}
\end{threeparttable}
\end{table}

\subsection*{Preprocessing of \textit{Pseudacris} sequence data}

Phased and aligned \textit{Pseudacris} sequence data was retrieved from Dryad
\citep{dryad_23rc0}. In the original analysis, an HKY nucleotide substitution
model was applied to 22 out of 26 nuclear loci \citep{Barrow201478}. To simply
our reanalysis, which was focused on performance and not reconstructing the
species tree \textit{per se}, we used only those 22 loci. To rank sampled
individual frogs by sequence quality, we counted the number of unambiguous base
calls for both haplotypes across all 22 HKY loci. To further avoid wasting of
computational resources, we then reduced the sequence data to both haplotypes
from the best-sequenced individual from each of the 19 ingroup extant lineages
in \cite{Barrow201478}.

\subsection*{StarBEAST2 reanalysis of \textit{Pseudacris} sequence data}

For inference of \textit{Pseudacris} trees, we ran 32 independent StarBEAST2
chains for each condition. Each chain used the same sequence data, but was an
independent estimate of convergence because different random seeds were used to
initialize each chain.

A birth-death prior was used for the species tree and the net diversification
and exinction fraction hyperparameter were estimated. An inverse gamma prior was
used for per-branch constant population sizes with a shape fixed at 2.0 and the
mean population size hyperparameter was estimated. Relative per-locus clock
rates were estimated using a log-normal prior distribution with the mean fixed
at 1.0 in real space and the standard deviation hyperparameter was estimated. An
HKY+$\Gamma$ substitution model with four gamma rate categories was applied to
all loci and all base frequencies were estimated separately for each locus. The
HKY $\kappa$ and gamma shape parameters were estimated and shared across all loci.

The rate of the root species tree branch was estimated for species tree relaxed
clock analyses, but the rates of gene tree root branches were not estimated as
it does not affect the phylogenetic likelihood. The standard deviation of the
UCLN clock model was fixed at 0.16 and 0.32 for species tree and gene tree
relaxed clocks respectively. The number of UCLN rate categories was set to equal
the number of species tree branches for species tree relaxed clocks (i.e. 37)
and the number of gene tree branches except the root branch for gene tree
relaxed clocks (i.e. 74).

To ensure convergence of all chains, we ran each chain for an initial length of
$2^{23}$ states, sampling every $2^{10}$ states. ESS values were computed for
all logged statistics after discarding 12.5\% of state samples as burnin. Logged
statistics included the posterior probability, the multispecies coalescent
(probability of gene trees given the species tree), birth death prior
probability of the species tree, the phylogenetic likelihood, the net
diversification rate, the extinction fraction, the HKY $\kappa$ parameter, the
among-site rate variation $\alpha$ parameter, the standard deviation of per-
locus clock rates, the mean population size, and the height of the species tree.

If any logged statistic had an ESS of below 200, the chain was resumed until the
length of the chain was doubled. ESS values were then re-evaluated, again after
discarding 12.5\% of state samples. The length of the chain was doubled and ESS
values re-evaluated until the ESS values of all logged statistics were above
200. The rate at which trees and statistics were sampled was halved with every chain
doubling so that the total number of samples remained constant.

ESS per hour was calculated by dividing the final ESS value for a given
statistic by 87.5\% of the total CPU time used by that chain to account for
burnin. Likewise ESS per million states was calculated by dividing the final ESS
value by 87.5\% of the total number of the states in the chain.

\subsection*{Simulation of trees and sequence data}

All simulation parameters were chosen to be broadly similar to those observed in
or estimated from the \textit{Pseudacris} data set.

First, 64 species trees were simulated according to a birth-death process
\citep{Gernhard2008769} using biopy with 19 extant species, a
speciation rate of 200.0 and a death rate of 50.0. This corresponds to a net
diversification rate of 150.0 and an extinction fraction of 0.25. Haploid
effective population sizes for each branch were chosen independently from an
inverse gamma distribution with a shape of 2.0 and a scale of 0.002. For a
species with annual generation times, including \textit{Pseudacris ornata} and
\textit{P. nigrita} \citep{10.2307/1446044}, this corresponds to a diploid
effective population size of 1000 individuals per generation. Species branch
rates were chosen from a log-normal distribution with a mean in real space of 1.0
and a standard deviation of 0.16, then scaled so that the mean of the branch
rates for a given species tree was exactly 1.0.

For each species tree, 220 gene trees with two sampled individuals per species
were simulated according to the multispecies coalescent process using biopy. The
mean clock rate for each locus was chosen from a log-normal distribution with a
mean in real space of 1.0 and a standard deviation of 0.3. The clock rates of
each sequential block of 22 loci were then scaled so that the mean of the clock
rates for a given block was exactly 1.0. Scaling branch rates and clock rates
was done to ensure that the accuracy of StarBEAST2 and concatenation reflected
relative rather than absolute differences between simulated and estimated times
and rates.

For each gene tree, 400nt long sequence alignments were simulated using Seq-Gen
\citep{Rambaut01061997}. An HKY model was used for all sequence alignments with
a transition/transversion $\kappa$ value of 3.0, and a discretized gamma model
of among-site rate variation with a shape $\alpha$ value of 0.2
\citep{Yang1994}.

\subsection*{Performance and accuracy of StarBEAST2 and concatenation}

The settings used for StarBEAST2 analyses of simulated sequence data were
identical to the analysis of \textit{Pseudacris} sequence data, except that 100 rate
categories were used for all UCLN relaxed clocks. Concatenation used the same
settings as StarBEAST2, but instead of inferring each gene tree within a species
tree, a single concatenated tree was inferred using the likelihoods of all loci.
The same strategy to ensure convergence was used as for \textit{Pseudacris}
analyses, but slightly modified for concatenated analyses; population sizes and
multispecies coalescent probabilities were not logged because they are not
estimated, and the initial chain length was only $2^{20}$ states, sampling every
$2^{7}$ states. ESS rates were calculated the same way as for
\textit{Pseudacris} analyses.

Relative species tree error is based on ``rooted branch score''
\citep[RBS;][]{Heled2013}. Given two trees $T_1$ and $T_2$, the sets of
monophyletic clades $c$ present in each tree are defined as $\mathbb{C}_1$ and
$\mathbb{C}_2$. The length of the parent branch extending from the root of the
subtree defined by $c$ is then $b(c)$. To calculate the rooted branch score, sum
all absolute differences in $b(c)$ between trees $T_1$ and $T_2$:

\begin{equation}
RBS(T_1, T_2) = \sum_{c \in {\mathbb{C}_1} \cup {\mathbb{C}_2}} |b^{(1)}(c) - b^{(2)}(c)|
\end{equation}

If a clade $c$ is present in only one of $T_1$ or $T_2$, then the branch length
$b(c)$ from the tree containing $c$ is added to the RBS. Relative species tree
error is the mean RBS between an estimated species tree and the true
species tree over the posterior distribution of species trees, and is normalized
by dividing by the total length of the true species tree
\citep{Ogilvie01052016}.

Pendant edge bias is defined as $\nicefrac{\hat{h} - h}{h}$, where the estimated
age for each extant species is $\hat{h}$ and the true age is $h$. The mean
pendant edge bias is the average pendant edge bias for all extant species in
every posterior sample.

Rooted Robinson-Foulds distances \citep{ROBINSON1981131} are defined as the
count of clades present only one of $T_1$ and $T_2$. In this study, the mean
rooted Robinson-Foulds distance is the average of all distances between each
estimated tree and the true tree in a posterior sample.

\section*{Supplementary Material}

Supplementary tables, figures, simulated data sets, and BEAST XML files will be
made available online.

\section*{Acknowledgments}

This work was supported by a Rutherford Discovery Fellowship awarded to AJD by
the Royal Society of New Zealand. HAO was supported by an Australian Laureate
Fellowship awarded to Craig Moritz by the Australian Research Council
(FL110100104). This research was undertaken with the assistance of resources
from the National Computational Infrastructure (NCI), which is supported by the
Australian Government. We wish to thank Jason Bragg and Renee Catullo for
testing StarBEAST2 before its official release, Tim Vaughan for suggesting the
addition of the root height changing operator, Joseph Heled for insight into the
multispecies coalescent model, and Graham Jones for input regarding operator
performance.

\clearpage

\bibliographystyle{natbib}
\bibliography{starbeast2}

\end{document}
